### PR TITLE
feat: add name, email, & dob

### PR DESCRIPTION
**Pull request by abutterworth**
- No live data yet
- Includes redux pipes and stub functions for api calls

### CODE REVIEW

File: src/account-settings/AccountSettingsPage.jsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl, intlShape } from 'react-intl';
+
+import messages from './AccountSettingsPage.messages';
+
+// Actions
```
Comment (albemarle) [inline src/account-settings/AccountSettingsPage.jsx]: Are these comments necessary?  It seems pretty obvious from the code.  (Plus selectors doesn't get a comment?  Not that I think it should.  :-) )

File: src/account-settings/AccountSettingsPage.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+AccountSettingsPage.defaultProps = {
+  loading: false,
+  loaded: false,
+  loadingError: null,
+  openFormId: null,
+  fieldSections: [
+    {
+      sectionHeading: 'Account Information',
```
Comment (albemarle) [inline src/account-settings/AccountSettingsPage.jsx]: i18n?

File: src/account-settings/AccountSettingsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        },
+        {
+          name: 'email',
+          label: 'Email address',
+          type: 'email',
+          helpText: 'You receive messages from Your Platform Name Here and course teams at this address.',
+        },
+        {
+          name: 'yearofbirth',
+          label: 'Year of birth',
```
Comment (albemarle) [inline src/account-settings/AccountSettingsPage.jsx]: Just a note when we do messages for real, Full Name is caps for each word, so probably Year of birth should be also.

File: src/account-settings/AccountSettingsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          type: 'select',
+          options: (() => {
+            const currentYear = new Date().getFullYear();
+            const years = [];
+            let startYear = currentYear - 120;
```
Comment (albemarle) [inline src/account-settings/AccountSettingsPage.jsx]: Optimistic, I like it.

File: src/account-settings/components/temp/Input.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  getHTMLTagForType() {
+    const { type } = this.props;
+    if (type === 'select' || type === 'textarea') return type;
+    return 'input';
+  }
+
+  getClassNameForType() {
+    const { type } = this.props;
+    if (type === 'file') return 'form-control-file';
```
Comment (albemarle) [inline src/account-settings/components/temp/Input.jsx]: Might be nicer as a switch?

File: src/account-settings/components/temp/Input.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    if (type === 'checkbox') return 'form-check-input';
+    if (type === 'radio') return 'form-check-input';
+    return 'form-control';
+  }
+
+  getRef(forwardedRef) {
+    if (process.env.NODE_ENV !== 'development') return forwardedRef;
```
Comment (albemarle) [inline src/account-settings/components/temp/Input.jsx]: Do we do this frequently?  I wonder if it's worth having a util function for isProd().

File: src/account-settings/components/temp/Input.jsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+Input.defaultProps = {
+  className: undefined,
+  options: [],
+};
+
+
+// Using React.forwardRef – more on forwarding refs here:
+// https://reactjs.org/docs/forwarding-refs.html
```
Comment (albemarle) [inline src/account-settings/components/temp/Input.jsx]: This comment would be more useful as "forwardRef floobs the wiginator in a crobley fashion".  I don't think the URL adds much; I am guessing Google will continue to exist at least as long as this code does.  ;-)

File: src/account-settings/reducers.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        loadingError: null,
+      };
+
+    case OPEN_FORM:
+      return {
+        ...state,
+        openFormId: action.payload.formId,
+      };
+    case CLOSE_FORM:
+      // Only close if the field to close is undefined or matches the field that is currently open
```
Comment (albemarle) [inline src/account-settings/reducers.js]: Could this comment be eliminated with a variable name?

File: src/account-settings/AccountSettingsPage.jsx (hunk 3 above)
Comment (abutterworth) [reply src/account-settings/AccountSettingsPage.jsx]: Oh it looks like our pattern is the opposite (Sentence case for labels). Will update.

File: src/account-settings/components/temp/Input.jsx (hunk 6 above)
Comment (davidjoy) [reply src/account-settings/components/temp/Input.jsx]: This exists - it's in configuration as `ENVIRONMENT`.  The configuration is available in redux.  

In general we should use the configuration object and not process.env to keep a layer of abstraction.

File: src/account-settings/components/temp/Input.jsx (hunk 5 above)
Comment (abutterworth) [reply src/account-settings/components/temp/Input.jsx]: Good call. This component is a copy of one in a PR for Paragon. I'll update there.

File: src/account-settings/components/temp/Input.jsx (hunk 6 above)
Comment (abutterworth) [reply src/account-settings/components/temp/Input.jsx]: This component is a copy of one in a PR for Paragon. In Paragon, I think `process.env.NODE_ENV` is the safest thing to use. What do you think?

**Pull request merged**